packages/dd-trace/src/plugins/util/git.js

```diff
@@ -15,8 +15,15 @@ const {
   CI_WORKSPACE_PATH
 } = require('./tags')
 
+const cache = new Map()
+
 // If there is ciMetadata, it takes precedence.
 function getGitMetadata (ciMetadata) {
+  const cacheKey = JSON.stringify(ciMetadata)
+  if (cache.has(cacheKey)) {
```

Comment:
It would be more efficient to `get` right away and check if there is a value to avoid 2 lookups.